### ISSUE TITLE
npm ERR! code ELIFECYCLE npm ERR! errno 1 after updating strapi

### ISSUE THREAD

**Issue by catalina-cimpanu** [strapi/strapi#8448]
Hello 👋 
I am writing because I have this problem, after updating the strapi project. I am using strapi as cms for gatsby. It was working fine before I had the wonderful idea of running `npm update` :(
As far as I understood from my research online, this might not be related to strapi, but since none of the recommended solutions worked for me, I am opening this issue here. 
I've been into web development only for a few months, so please be kind.

## Bug report
So, after updating `npm update` , when I ran `npm run develop` , the server wouldn't start, and I got the following error:

```
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! yn-strapi@0.1.0 develop: `strapi develop`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the yn-strapi@0.1.0 develop script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\catal\AppData\Roaming\npm-cache\_logs\2020-10-22T20_28_05_373Z-debug.log
```
I of course checked online, and I found this post [https://stackoverflow.com/questions/42308879/how-to-solve-npm-error-npm-err-code-elifecycle](https://stackoverflow.com/questions/42308879/how-to-solve-npm-error-npm-err-code-elifecycle)
but I tried all the recommended solutions, but it still doesn't work.

I tried:

1.  `$ npm cache clean --force` then `$ rm -rf node_modules package-lock.json` then `npm install`  - didn't work
2.  `npm cache clean --force` then deleted node_modules and package_lock.json then `npm install` - didn't work
3.  I deleted everything manually then ran  `npm install`   - didn't work
4.  I reinstalled/updated npm with `npm install npm` - didn't work
5.  I installed strapi globally - didn't work
6.  Even tried the very short `rm -rf node_modules && rm ./package-lock.json && npm install` - didn't work
7.  Also deleted only the package_lock.json, then ran `npm install`  - didn't work
8.  Restarted my computer - didn't work

### Steps to reproduce the behavior
I don't know how to reproduce it, because I created a new strapi app, and inside this new app, everything works fine (which makes it even more confusing).
I also don't want to rewrite everything, it would be quite cumbersome to set up everything once again.

### Expected behavior
npm run develop works

### Code snippets
This is from the console:
```
catal@Catalina MINGW64 ~/Catalina/Learning stuff/Web development/young neuros barebone remake/young-neuros-gatsby-strapi/yn-strapi (master)
$ npm run develop

> yn-strapi@0.1.0 develop C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi
> strapi develop

[2020-10-22T20:50:10.897Z] debug Γ¢ö∩╕Å Server wasn't able to start properly.
[2020-10-22T20:50:10.898Z] error TypeError: Cannot read property 'includes' of undefined
    at Object.isPrivateAttribute (C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\node_modules\strapi-utils\lib\content-types.js:69:34)
    at C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\services\type-definitions.js:49:24
    at apply (C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\node_modules\lodash\lodash.js:475:27)
    at wrapper (C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\node_modules\lodash\lodash.js:5059:16)
    at C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\services\type-definitions.js:177:28
    at Array.filter (<anonymous>)
    at buildAssocResolvers (C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\services\type-definitions.js:177:6)
    at buildCollectionType (C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\services\type-definitions.js:399:12)
    at C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\services\type-definitions.js:286:16
    at Array.map (<anonymous>)
    at buildModels (C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\services\type-definitions.js:275:17)
    at buildModelsShadowCRUD (C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\services\schema-generator.js:133:8)
    at Object.generateSchema (C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\services\schema-generator.js:29:42)
    at Object.initialize (C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi-plugin-graphql\hooks\graphql\index.js:81:9)
    at C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\strapi\lib\hooks\index.js:37:28
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! yn-strapi@0.1.0 develop: `strapi develop`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the yn-strapi@0.1.0 develop script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\catal\AppData\Roaming\npm-cache\_logs\2020-10-22T20_50_10_920Z-debug.log
```

This is the log (inside the log file):
```
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Users\\catal\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'develop'
1 verbose cli ]
2 info using npm@6.14.8
3 info using node@v12.18.3
4 verbose run-script [ 'predevelop', 'develop', 'postdevelop' ]
5 info lifecycle yn-strapi@0.1.0~predevelop: yn-strapi@0.1.0
6 info lifecycle yn-strapi@0.1.0~develop: yn-strapi@0.1.0
7 verbose lifecycle yn-strapi@0.1.0~develop: unsafe-perm in lifecycle true
8 verbose lifecycle yn-strapi@0.1.0~develop: PATH: C:\Users\catal\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi\node_modules\.bin;C:\Users\catal\bin;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\local\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Users\catal\bin;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0;C:\windows\System32\OpenSSH;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\Calibre2;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0;C:\WINDOWS\System32\OpenSSH;C:\Program Files\Git\cmd;C:\Program Files\nodejs;C:\Users\catal\AppData\Local\Programs\Python\Python38-32\Scripts;C:\Users\catal\AppData\Local\Programs\Python\Python38-32;C:\Users\catal\AppData\Local\Microsoft\WindowsApps;C:\Users\catal\AppData\Local\rodeo\app-2.5.2\bin;C:\Users\catal\AppData\Local\Programs\MiKTeX\miktex\bin\x64;C:\Users\catal\AppData\Local\hyper\app-3.0.2\resources\bin;C:\Users\catal\AppData\Roaming\npm;C:\Program Files\heroku\bin;C:\Users\catal\AppData\Local\Microsoft\WindowsApps;C:\Users\catal\AppData\Local\Programs\Microsoft VS Code\bin;C:\Program Files\Git\usr\bin\vendor_perl;C:\Program Files\Git\usr\bin\core_perl
9 verbose lifecycle yn-strapi@0.1.0~develop: CWD: C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi
10 silly lifecycle yn-strapi@0.1.0~develop: Args: [ '/d /s /c', 'strapi develop' ]
11 silly lifecycle yn-strapi@0.1.0~develop: Returned: code: 1  signal: null
12 info lifecycle yn-strapi@0.1.0~develop: Failed to exec develop script
13 verbose stack Error: yn-strapi@0.1.0 develop: `strapi develop`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Users\catal\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:315:20)
13 verbose stack     at ChildProcess.<anonymous> (C:\Users\catal\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:315:20)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
14 verbose pkgid yn-strapi@0.1.0
15 verbose cwd C:\Users\catal\Catalina\Learning stuff\Web development\young neuros barebone remake\young-neuros-gatsby-strapi\yn-strapi
16 verbose Windows_NT 10.0.19041
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\catal\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "develop"
18 verbose node v12.18.3
19 verbose npm  v6.14.8
20 error code ELIFECYCLE
21 error errno 1
22 error yn-strapi@0.1.0 develop: `strapi develop`
22 error Exit status 1
23 error Failed at the yn-strapi@0.1.0 develop script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

### System

- Node.js version: v12.18.3
- NPM version: 6.14.8
- Strapi version: 3.2.5
- Database:  mongoDB 
- Operating system:  Windows 10

### Additional context

Would be very happy if you can help, it would save me some time, otherwise, I'll probably just redo everything again.

**Comment by derrickmehaffy** [comment]
`npm update` Please do not use this command to update Strapi projects, in your case the issue most likely is related to our [Migration Guides](https://strapi.io/documentation/v3.x/migration-guide/)

We also recommend that you manually update the package versions following this guide: https://strapi.io/documentation/v3.x/guides/update-version.html

Please do ensure that none of your Strapi packages have `^` before the version as this could cause a conflict where one package is updated beyond another.

**Comment by catalina-cimpanu** [comment]
I see. Thank you very much for your answer. 

As I said, the new strapi worked fine, so initially I tried to copy the files from the api folder from the old strapi to the new strapi, and that's when it stopped working. Then I saw that the structure is slightly different in the new strapi, and I thought that's why it broke. 
In the end, I actually stuck to the newly created strapi app, and I connected it to the database I already had. I had only to redo the schemas in strapi, but not reenter all the data, which is good :)

Anyway, I didn't click on the migrating problem, but very probably it's because of that. I would have anyway wanted the new version, but I could have done the migration with fewer headaches :))

Thanks!
